File: ruby/pangram/pangram.rb
class Pangram

  def self.pangram? sentence
    string   = sentence.to_s
    ('a'..'z').all? { |letter| string.downcase.include?(letter) }

    # return false if string.eql? ""
    #
    # # make a hash of all letters with their count
    # tracker = Hash[ ('a'..'z').collect { |letter| [letter, 0] } ]
    #
    # # downcase string into an array of characters & filter out non letters
    # letters = string.downcase.scan /[a-z]/
    #
    # # loop through each letter and add one to the tracker for each letter
    # letters.each { |l| tracker[l] += 1 }
    #
    # # are all letters non-zero in the hash
    # tracker.none? { |letter,count| count == 0 }
  end

end

File: elixir/pangram/pangram.exs
defmodule Pangram do

  @alphabet ( ?a..?z |> Enum.map( &Kernel.to_string([&1]) ) )
  # @alphabet [ "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m",
  #             "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"]

  @spec pangram?(String.t()) :: boolean
  def pangram?(""), do: false
  def pangram?(sentence) do
    letters  = sentence |> String.replace(~r/[\s\d\W]/, "")
                        |> String.downcase |> String.graphemes
                        |> Enum.sort |> Enum.uniq
    # is every letter of the alphbet found in list of letters of the sentence
    Enum.all?(@alphabet, &(&1 in letters))
  end

end
